pina/operator.py

```diff
@@ -328,103 +272,151 @@ def laplacian(output_, input_, components=None, d=None, method="std"):
     return LabelTensor(result, labels=labels)
 
 
-def _scalar_laplacian(output_, input_, d):
+def fast_advection(output_, input_, velocity_field, components, d):
     """
-    Compute the laplacian of a scalar-valued ``output_``.
+    Perform the advection operation on the ``output_`` with respect to the
+    ``input``. This operator support vector-valued functions with multiple input
+    coordinates.
 
-    :param LabelTensor output_: The output tensor on which the laplacian is
-        computed. It must be a column tensor.
+    Unlike ``advection``, this function performs no internal checks on input and
+    output tensors. The user is required to specify both ``components`` and
+    ``d`` as lists of strings. It is designed to enhance computation speed.
+
+    :param LabelTensor output_: The output tensor on which the advection is
+        computed.
+    :param LabelTensor input_: the input tensor with respect to which advection
+        is computed.
+    :param str velocity_field: The name of the output variable used as velocity
+        field. It must be chosen among the output labels.
+    :param list[str] components: The names of the output variables for which to
+        compute the advection. It must be a subset of the output labels.
+    :param list[str] d: The names of the input variables with respect to which
+        the advection is computed. It must be a subset of the input labels.
+    :return: The computed advection tensor.
+    :rtype: torch.Tensor
+    """
+    # Save the velocity field
+    velocity = output_.extract(velocity_field)
+
+    # Remove the velocity field from the components
+    filter_components = [c for c in components if c != velocity_field]
+
+    tmp = (
+        fast_grad(
+            output_=output_, input_=input_, components=filter_components, d=d
+        )
+        .reshape(-1, len(filter_components), len(d))
```

Comment:
I am not sure we really want to do this reshape, what if the input is 3D? At the end we are left with a 2D output